docs/tutorial.md

```diff
@@ -3,200 +3,490 @@
 
-Here we're listening for [`playerJoined`](http://mineflayer.prismarine.js.org/#/api?id=quotplayerjoinedquot-player) event.
-And sending welcome message to chat.
+If you want to do something for every item in an array, a `for in` or `for of` loop can be useful.  
 
 ```js
-function joined (player) {
-  bot.chat(`Welcome ${player}! :D`)
+const array = [1, 2, 3]
+
+for (const index in array) {
+  console.log(array[index])
 }
 
-bot.on('playerJoined', joined)
+for (const item of array) {
+  console.log(item)
+}
 ```
 
-### Callbacks
+A `for in/of` loop needs to have a variable before the in/of, this is the variable that can be used to access the current item.  
+The variable after the in/of needs to be something that contains other variable. These are mostly arrays, but also some objects.  
+A `for in` loop uses the index of each item, while the `for of` loop uses the items directly.  
+Which one is useful depends on what you're going to do with it.
 
-Callback (known as a "call-after" function) is a function that is passed as an argument to the other code / function.
-If that other code / function finish the execution, then the callback is called.
+#### General tips
 
-Callback takes parameter called `err` or `error` as first, and second parameter (if specified) is a value that is passed to the callback,
-if `err === null` then there is no error happened in the other code / function.
+If you want to help yourself and other people understand your code better, you can use comments.  
+Comments can be created using `//` and everything after that is completely ignored by Javascript.
 
-In [the API page](http://mineflayer.prismarine.js.org/#/api) callback abbreviated as `cb` or `callback`.
+In general, it is good practice to use the `const` keyword instead of the `let` keyword when defining a variable. A variable defined with `const` can't be modified later and is thus constant.  
+Javascript is then able to make your code run more efficiently because it knows it doesn't have to account for variable changes for that variable.  
+If you want a modifiable variable, you will still have to use `let` of course.
 
-Examples :
+```js
+const test = 5
+test = 10 // This line is not valid, as test is constant.
+```
 
-#### Craft Stick Bot
+#### Node Package manager
 
-Here we're crafting oak logs into sticks.
-Craft oak logs into oak wood planks, wait for crafting finish. Then craft the oak wood planks into sticks.
+The last thing you need to know is how to use the [Node Package Manager](https://www.npmjs.com/).  
+NPM is automatically installed when you install Node.  
+NPM is used to get useful packages that other people created that can do useful things for you.  
+You can search for packages on [their website](https://www.npmjs.com/), and then install them using the `npm install` command in your terminal.  
+To install Mineflayer for example, run `npm install mineflayer`  
 
-Incorect aproach ❌ :
```

Comment:
any reason to remove this part ? I thought it was useful

Reply:
Hmmm, I agree with you. I guess when I reorganized/restructured the tutorial, I removed a bit too much.
I'll add something like it back.